### ISSUE TITLE
Ctrl+Left/Right moves differently than Ctrl+Shift+Left/Right

### ISSUE THREAD

**Issue by Davidebyzero** [notepad-plus-plus/notepad-plus-plus#5144]
### Description of the Issue
Moving the cursor a "word" at a time using Ctrl+Left or Ctrl+Right should act exactly the same as doing so with Ctrl+Shift+Left or Ctrl+Shift+Right, respectively, except that the latter pair both moves and selects, while the former only moves. However, the latter move differently than the former. As it so happens, I much prefer the movement logic of the former.

### Steps to Reproduce the Issue
1. On an otherwise empty line by itself, put:
<pre>(((        )))</pre>
2. Move the caret (text cursor) to the beginning of that line
3. Press Ctrl+Right twice. The caret first moves to the beginning of the next group of `(((`, and then to the end of that group.
4. Press Home to go back to the beginning of the line.
5. Press Ctrl+Shift+Right. The caret will be moved only to the end of the first `(((`, with exactly the first `(((` being selected. This is completely different than the Ctrl+Right behavior, but it might actually be better. (If the decision is made that this is the better way, the Ctrl+Right behavior should be changed to match. Otherwise, the Ctrl+Shift+Right behavior should be changed to match.)
6. Press Ctrl+Shift+Right again. The caret will be moved to the end of the last `(((`, without making a stop at the beginning of the last `(((`. This is the worst thing about this Ctrl+Shift+Left/Right behavior; what I would always want to do in this situation is select up to but not including the beginning of the last `(((`, and the actual behavior of Notepad++ in this situation throws me off.

### Expected Behavior
Ctrl+Left/Right should move with the same logic regardless of whether Shift is held. The only difference that holding Shift should make is to enable selection during that movement.

### Actual Behavior
Already described in "Steps to Reproduce".

### Further Information
Tuning the exact behavior of Ctrl+Left/Right move-one-word-at-a-time to work as desired could significantly enhance my productivity. Many applications implement this in what I believe to be the wrong way, and I find that it slows me down. But I don't recall ever coming across an app that makes it behave differently when Shift is held. Whichever move-one-word-at-a-time behavior would be best is up for discussion (and it should probably be configurable), but it should definitely be the same whether Shift is held or not.

**Comment by Davidebyzero** [comment]
Just found out that this can be fixed in Settings → Shortcut Mapper → Scintilla commands. The default is for Ctrl+Shift+Left to be bound to SCI_WORDLEFTENDEXTEND; removing this mapping and replacing it with SCI_WORDLEFTEXTEND (and likewise for Ctrl+Shift+Right) has the desired effect.

I would still consider this to be a bug in the default mapping.

**Comment by sasumner** [comment]
@Davidebyzero 

I looked in Notepad++ 7.6 and I see a different default keymapping:

SCI_WORDLEFT ctrl+left
SCI_WORDRIGHT ctrl+right
SCI_WORDLEFTEXTEND ctrl+shift+left
SCI_WORDRIGHT**END**EXTEND ctrl+shift+right

This is what seems to lead to the original inconsistent behavior.  I wonder if it is intended for some reason unknown to us at the moment?

**Comment by Davidebyzero** [comment]
Oh, I guess I wasn't paying attention. Ctrl+Shift+Right is what I had been trying to use lately, and failing each time because I expected SCI_WORDRIGHTEXTEND behavior but was getting SCI_WORDRIGHT*END*EXTEND behavior instead.

When I saw today that I could fix this by editing the keymapping, I deleted both mappings of Ctrl+Shift+Left/Right and then recreated them using the non-END versions. I didn't actually check to see if Ctrl+Shift+Left was bound to SCI_WORDLEFT*END*EXTEND before.

I just did a clean installation of Notepad++ in a VM and you're right, only Ctrl+Shift+Right defaults to the END version. So that is doubly inconsistent, and hopefully a bug rather than intended.
